src/Abstractions/NexusMods.Abstractions.Downloads/ADownloadActivity.cs

```diff
@@ -0,0 +1,77 @@
+using JetBrains.Annotations;
+using NexusMods.Abstractions.Activities;
+using NexusMods.MnemonicDB.Abstractions;
+using NexusMods.Paths;
+using ReactiveUI;
+using ReactiveUI.Fody.Helpers;
+
+namespace NexusMods.Abstractions.Downloads;
+
+/// <summary>
+/// Implementation of <see cref="IDownloadActivity"/> for ease of use.
+/// </summary>
+[PublicAPI]
+public abstract class ADownloadActivity : ReactiveObject, IDownloadActivity
+{
+    /// <inheritdoc/>
+    public PersistedDownloadStateId PersistedStateId { get; }
+
+    /// <inheritdoc/>
+    public IDownloader Downloader { get; }
+
+    /// <inheritdoc/>
+    public string Title { get; }
+
+    /// <inheritdoc/>
+    public AbsolutePath DownloadPath { get; }
+
+    /// <summary>
+    /// Constructor.
+    /// </summary>
+    protected ADownloadActivity(
+        PersistedDownloadState.ReadOnly persistedState,
+        IDownloader downloader,
+        AbsolutePath downloadPath)
+    {
+        PersistedStateId = persistedState.PersistedDownloadStateId;
+        Status = persistedState.Status;
+
+        Downloader = downloader;
+        Title = persistedState.Title;
+        Downloader = downloader;
+    }
+
+    /// <inheritdoc/>
+    public PersistedDownloadStatus Status { get; private set; }
```

Comment:
This still assumes all status changes go through here. 
Currently the pattern we use is to instead listen to the DB for changes since db can be changed from anywhere in the code.

There is no good way to define this as the sole authority for Status, especially since the actual agents of the changes would be the IDownloaders.

I wouldn't mind having better separation of concerns and more controlled write access to DB data, but I don't think we have any way to express this or enforce it, so it would be easy for this to be forgotten and bugs to come from it.